src/sentry/search/events/builder.py

```diff
@@ -0,0 +1,272 @@
+from datetime import datetime
+from typing import Any, Callable, List, Mapping, Optional, Sequence, Union
+
+from parsimonious.exceptions import ParseError
+from snuba_sdk.column import Column
+from snuba_sdk.conditions import BooleanCondition, Condition, Op, Or
+from snuba_sdk.entity import Entity
+from snuba_sdk.expressions import Limit
+from snuba_sdk.function import CurriedFunction, Function
+from snuba_sdk.orderby import Direction, OrderBy
+from snuba_sdk.query import Query
+
+from sentry.api.event_search import SearchFilter, SearchKey, SearchValue, parse_search_query
+from sentry.exceptions import InvalidSearchQuery
+from sentry.search.events.constants import (
+    EQUALITY_OPERATORS,
+    NO_CONVERSION_FIELDS,
+    TAG_KEY_RE,
+    VALID_FIELD_PATTERN,
+)
+from sentry.search.events.fields import FIELD_ALIASES, is_function
+from sentry.utils.snuba import Dataset, resolve_column
+
+# TODO: this should be a TypedDict instead
+ParamsType = Mapping[str, Union[List[int], int, str, datetime]]
+AggregateType = Union[Function, CurriedFunction]
+WhereType = Union[Condition, BooleanCondition]
+SelectType = Union[Column, CurriedFunction, Function]
+
+OPERATOR_MAP: Mapping[str, Op] = {
```

Comment:
You could cast the operators instead of maintaining this map. `Op("=") == Op.EQ`

Reply:
_whoaa_ nice 👏